api/src/utils/validate-donation.test.ts

```diff
@@ -0,0 +1,21 @@
+import { inLastFiveMinutes } from './validate-donation';
+
+describe('inLastFiveMinutes', () => {
+  it('should return true if the timestamp is within the last five minutes', () => {
+    const currentTimestamp = Math.floor(Date.now() / 1000);
+    const recentTimestamp = currentTimestamp - 100; // 100 seconds ago
+    expect(inLastFiveMinutes(recentTimestamp)).toBe(true);
+  });
+
+  it('should return false if the timestamp is more than five minutes ago', () => {
+    const currentTimestamp = Math.floor(Date.now() / 1000);
+    const oldTimestamp = currentTimestamp - 400; // 400 seconds ago
+    expect(inLastFiveMinutes(oldTimestamp)).toBe(false);
+  });
+
+  it('should return true if the timestamp is exactly five minutes ago', () => {
+    const currentTimestamp = Math.floor(Date.now() / 1000);
+    const exactTimestamp = currentTimestamp - 300; // 300 seconds ago
+    expect(inLastFiveMinutes(exactTimestamp)).toBe(true);
+  });
```

Comment:
This looks like it might be flaky, since there's a small chance that enough time has passed for `timeDifference` to be greater than 300.

To avoid flake, I recommend using fake timers. We have some examples of that in api/src/utils/tokens.test.ts